### PR TITLE
TECH-474: Update BB Digital Registry - GET /data/MyPersonalDataUsage/1.0

### PR DESCRIPTION
**Pull request by KarolinaKopacz**
Update BB Digital Registry - GET /data/MyPersonalDataUsage/1.0

Ticket: https://govstack-global.atlassian.net/browse/TECH-474

### CODE REVIEW

File: examples/mock/mockoon-digitalregistries.json [hunk 1]
```diff
@@ -14,8 +14,8 @@
       "endpoint": "data/MyPersonalDataUsage/1.0",
       "responses": [
         {
-          "uuid": "af65df73-2417-459e-8156-725979689d70",
-          "body": "[\n  {\n    \"ID\": \"1234567\",\n    \"ReaderID\": \"{{queryParam 'userID'}}\",\n    \"ReaderInitials\": \"JD\",\n    \"ReaderInstitutionID\": \"EE70049837\",\n    \"ReaderInstitutionName\": \"East Hospital\",\n    \"ReaderApplicationName\": \"East Hospital healthcare back office\",\n    \"SearchDateTime\": \"2017-07-21T17:32:28Z\",\n    \"Refrences\": [\n      {\n        \"ReferenceID\": \"{{queryParam 'DatabaseID'}}\"\n      }\n    ]\n  }\n]",
+          "uuid": "ad5ddc6e-bd4f-491b-b600-5cab337701f5",
+          "body": "[\n  {\n    \"ID\": \"1234567\",\n    \"ReaderID\": \"{{queryParam 'userID'}}\",\n    \"ReaderInitials\": \"JD\",\n    \"ReaderInstitutionID\": \"EE70049837\",\n    \"ReaderInstitutionName\": \"East Hospital\",\n    \"ReaderApplicationName\": \"East Hospital healthcare back office\",\n    \"SearchDateTime\": \"2017-07-21T17:32:28Z\",\n    \"Refrences\": [\n      {\n        \"ReferenceID\": \"MCTS311\"\n      }\n    ]\n  }\n]",
```
Comment (mgrochalskisoldevelo) [inline examples/mock/mockoon-digitalregistries.json:18]: I think 
```
ReaderID
```
won't be the same as this one from the request. 

Request -> User's personal unique identifier (User that wants to see who has read his personal data)
Response -> Identifier of a reader who looked at personal data

File: test/openAPI/features/data_my_personal_data_usage.feature [hunk 2]
```diff
@@ -1,17 +1,48 @@
...
-    Then The user receives a list of all records
+The purpose of this API is to make personal data protection better and make BB personal 
+data usage transparent by showing who has looked at personal data of the user.
 
-  Scenario: The user receives an empty list of records that have read his personal data from the Digital Registries database
-    Given The user wants to check who has read his personal data from Digital Registries database and there is an empty list of users who have done so
-    When The user sends a valid request and receives a list of users who have read his personal data
-    Then The user receives an empty list because there are no records in the database about other users who have read his personal data
+  @smoke @unit @positive
+  Scenario: The user gets a list of all records that have read their personal data smoke test type
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/data_my_personal_data_usage.feature:8]:
```suggestion
  Scenario: The user gets a list of all records that have read his personal data smoke test type
```

File: test/openAPI/features/data_my_personal_data_usage.feature [hunk 3]
```diff
@@ -1,17 +1,48 @@
...
+    And The /data/MyPersonalDataUsage/1.0 endpoint response should have status 200
+    And The /data/MyPersonalDataUsage/1.0 endpoint response should have content-type: application/json header
+    And The /data/MyPersonalDataUsage/1.0 endpoint response should match json schema
 
-  Scenario: The user is unable to obtain data from the Digital Registries database due to an invalid request
-    Given The user wants to check who has read his personal data in the Digital Registries database
-    When The user sends an invalid request and receives a list of users who have read his personal data
-    Then The operation to get a list of results is an error
+  @unit @positive
+  Scenario Outline: The user gets a list of all records that have read their personal data
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/data_my_personal_data_usage.feature:18]:
```suggestion
  Scenario Outline: The user gets a list of all records that have read his personal data
```

File: test/openAPI/features/support/data_my_personal_data_usage.js [hunk 4]
```diff
@@ -1,102 +1,121 @@
...
-    .get(baseUrl)
-    .withHeaders(`${header.key}`, `${header.value}`)
-    .withQueryParams('userID', userID)
-    .withQueryParams('DatabaseID', databaseID);
+const baseUrl = localhost + dataMyPersonalDataUsageEndpoint;
+const endpointTag = { tags: `@endpoint=/${dataMyPersonalDataUsageEndpoint}` };
 
-Before(() => {
+Before(endpointTag, () => {
   specDataMyPersonalDataUsage = pactum.spec();
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/support/data_my_personal_data_usage.js:23]:
```suggestion
  specDataMyPersonalDataUsage = spec();
```

File: test/openAPI/features/support/data_my_personal_data_usage.js [hunk 5]
```diff
@@ -1,102 +1,121 @@
...
 
-let userID;
-let databaseID;
-let specDataMyPersonalDataUsage;
+chai.use(require('chai-json-schema'));
 
-const baseUrl = `${localhost}data/MyPersonalDataUsage/1.0`;
+let specDataMyPersonalDataUsage;
+let returnedReaderId;
+let returnedReferanceArray;
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/support/data_my_personal_data_usage.js:17]: seems like this is not used

**Review by mgrochalskisoldevelo — CHANGES_REQUESTED** [review]
A few small comments from me. Also I think you missed one response 
image

**Review by mgrochalskisoldevelo — APPROVED** [review]
lgtm

**Review by ingmarvali — APPROVED** [review]

**Comment by mdulko** [comment]
lgtm

**Pull request merged**